test/tests/auxkernels/diffusion_flux/diffusion_flux.i

```diff
@@ -0,0 +1,95 @@
+[Mesh]
+  type = GeneratedMesh # Can generate simple lines, rectangles and rectangular prisms
+  dim = 2 # Dimension of the mesh
+  nx = 10 # Number of elements in the x direction
+  ny = 10 # Number of elements in the y direction
+  xmax = 1.0
+  ymax = 1.0
+[]
+
+[Variables]
+  [./T]
+  [../]
+[]
+
+[AuxVariables]
+  [./flux_x]
+      order = FIRST
+      family = MONOMIAL
+  [../]
+  [./flux_y]
+      order = FIRST
+      family = MONOMIAL
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = HeatConduction # A Laplacian operator
```

Comment:
You need to change this to use a Diffusion kernel

Reply:
Actually, use the MatDiffusion kernel, with the parameter 'prop_name = thermal_conductivity'.